plugins/Marketplace/Controller.php

```diff
@@ -318,17 +318,27 @@ public function installAllPaidPlugins()
 
         for ($i = 0; $i <= 10; $i++) {
             foreach ($paidPlugins as $index => $paidPlugin) {
+                if (empty($paidPlugin)) {
+                    continue;
+                }
+
                 $pluginName = $paidPlugin['name'];
 
                 if ($this->pluginManager->isPluginActivated($pluginName)) {
-                    unset($paidPlugins[$index]);
+                    // we do not use unset since it might skip a plugin afterwards when removing index
+                    $paidPlugins[$index] = null;
+                    continue;
+                }
+
+                if (!$this->pluginManager->isPluginInFilesystem($pluginName)) {
```

Comment:
Prevent trying to install a plugin that is not available in the filesystem. We tried to install or activate all of them.